### DISCUSSION TITLE
What are the Limits

### DISCUSSION THREAD

**Discussion by TheBrightSoul** [community/community#146740] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

**Greetings!** 🌟  

1. Is there a limit on the number of commits allowed in a single repository on GitHub or Git? For example, is it restricted after reaching 1000 commits?  
2. What is the maximum size limit for a GitHub repository? For instance, is there a restriction like repositories cannot exceed 200MB or a specific size in MB/GB?  

**Thank you in advance!** 😊

**Comment by Barmanji** [comment, marked as answer]
Limit on Commits in Git/GitHub

There is no limit for the number of commits you might do in a single repository on Git or GitHub. You may have thousands or millions of commits without any trouble. But the performance of operations such as cloning, searching history, etc will be slow in very large-commit repositories if the size of the repository is larger and histories are complex in nature.
**Maximum Size Limit of GitHub Repositories**

1. Recommended Repository Size:
The company recommends keeping all the repositories under 1 GB for high performance.
This should not be over 100 MB in size files. GitHub will even warn you if you try to push files of this size.
Large files store much better in [Git Large File Storage (LFS)](https://git-lfs.github.com/).

2. Hard Repository Limit:
-There is no enforcement on the total size of a repository, though if your repository exceeds **5 GB**, GitHub might reach out to you and request either reducing its size or switching to alternative hosting options, like Git LFS.

3. **Git Limitations:** The size of a repository is not limited by Git itself, but it can be very slow when handling large files or a really deep history.

Key Points to Large Repositories

In addition, it uses **Git LFS** to store large binary files. Clean out unused files or large histories with `git filter-repo` or use a garbage collector called `git gc` to optimize your repository. End-Only archive old, unused repositories instead of hosting everything in one. 

Have a good day! 
adios

**Reply by TheBrightSoul** [reply]
Got it thanks.

**Comment by harmeetsingh11** [comment]
Hi @TheBrightSoul 

### 1. **Commit Limit in GitHub or Git**

There is **no strict limit** on the number of commits allowed in a Git repository, and GitHub does not impose a specific cap on commits. You can make as many commits as you need. However, **performance** can degrade with a very high number of commits, especially for large repositories.

- **Git itself** is designed to handle a large number of commits efficiently, and repositories with thousands or millions of commits are common.
- **GitHub** may experience slower performance for repositories with numerous commits, particularly when interacting with the web interface or cloning large repositories.

As such, while there's no hard limit, performance can be impacted by the size and complexity of the repository.

### 2. **Maximum Size Limit for GitHub Repositories**

GitHub does have **recommended size limits** for repositories, especially when dealing with large files:

- **Repository Size Limit**: GitHub recommends keeping repositories under **1 GB** for optimal performance. While GitHub technically allows larger repositories, performance issues can arise once you exceed this sizeFile Size Limit: There is a **100 MB limit** for individual files in a repository. Files larger than this cannot be pushed directly to GitHub, and you'll need to use **Git LFS** (Large File Storage) for handling files over 100 MB  .

- **Git LFS** to store large files (over 100 MB) outside the regular Git repository, making it easier to manage large files in repositories without impacting performance. However, there are limits to Git LFS usage, with **1 GB of storage** and **1 GB of bandwidth per month** for free users, and additional costs for higher usage .

For further details on large files with GitHub, refer to the [GitHub documentation](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github).

**Reply by TheBrightSoul** [reply]
Got it thanks
